### PR TITLE
pre-commit: test PR88609 (New)

### PR DESCRIPTION
**Pull request by dtcxzyw**
PR Link: https://github.com/llvm/llvm-project/pull/88609

### CODE REVIEW

File: bench/assimp/optimized/DXFLoader.cpp.ll [hunk 1]
```diff
@@ -6732,36 +6732,38 @@ _ZNSt6vectorIjSaIjEE9push_backERKj.exit:          ; preds = %if.then.i, %_ZNSt6v
   %colors = getelementptr inbounds i8, ptr %24, i64 24
   %_M_finish.i134 = getelementptr inbounds i8, ptr %24, i64 32
   %_M_end_of_storage.i135 = getelementptr inbounds i8, ptr %24, i64 40
-  %93 = add nuw nsw i8 %86, %tobool.mask
-  %94 = add nuw nsw i8 %93, 2
-  %wide.trip.count = zext nneg i8 %94 to i64
+  %93 = and i8 %b.sroa.0.0.lcssa, 1
+  %94 = zext nneg i8 %93 to i32
```
Comment (dtcxzyw) [inline bench/assimp/optimized/DXFLoader.cpp.ll:6736]: Regression

File: bench/c3c/optimized/sema_decls.c.ll [hunk 2]
```diff
@@ -8947,11 +8947,10 @@ vec_erase_ptr_at.exit:                            ; preds = %vec_erase_ptr_at.ex
   br label %175
 
 175:                                              ; preds = %171, %167
-  %.mask = and i8 %spec.select152, 1
-  %176 = zext nneg i8 %.mask to i64
+  %176 = zext nneg i8 %spec.select152 to i64
   %177 = shl nuw nsw i64 %176, 13
   %178 = and i64 %147, -8193
-  %179 = or disjoint i64 %177, %178
+  %179 = or i64 %177, %178
```
Comment (goldsteinn) [inline bench/c3c/optimized/sema_decls.c.ll:8953]: Why are we unable to prove `disjoint` anymore?

File: bench/graphviz/optimized/acyclic.c.ll [hunk 3]
```diff
@@ -133,37 +133,39 @@ define internal fastcc zeroext i1 @dfs(ptr noundef %0, ptr noundef %1, i1 nounde
...
 50:                                               ; preds = %46
-  %51 = trunc i8 %.051 to i1
+  %51 = trunc nuw i8 %.051 to i1
   %52 = tail call fastcc zeroext i1 @dfs(ptr noundef %0, ptr noundef nonnull %24, i1 noundef zeroext %51, ptr noundef %3)
   %53 = zext i1 %52 to i8
   %54 = or i8 %.051, %53
-  br label %55
+  %55 = icmp ne i8 %54, 0
+  %56 = zext i1 %55 to i8
+  br label %57
```
Comment (goldsteinn) [inline bench/graphviz/optimized/acyclic.c.ll:151]: Regression.

File: bench/msdfgen/optimized/sdf-error-estimation.cpp.ll [hunk 4]
```diff
@@ -1288,7 +1291,8 @@ land.lhs.true227:                                 ; preds = %invoke.cont218
 if.then229:                                       ; preds = %land.lhs.true227
   %incdec.ptr.i250 = getelementptr inbounds i8, ptr %77, i64 -16
   store ptr %incdec.ptr.i250, ptr %_M_finish.i204, align 8
-  %frombool232 = xor i8 %inside.1.lcssa, 1
+  %lnot231 = and i8 %inside.1.lcssa, 1
+  %frombool232 = xor i8 %lnot231, 1
```
Comment (goldsteinn) [inline bench/msdfgen/optimized/sdf-error-estimation.cpp.ll:1295]: Regression here and above.

**Pull request closed without merging**